ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Runner.scala

```diff
@@ -153,15 +147,29 @@ final class Runner[T <: ReadWriteService, Extra](
                   .acquire()
                 healthChecksWithIndexer <- participantConfig.mode match {
                   case ParticipantRunMode.Combined | ParticipantRunMode.Indexer =>
-                    new StandaloneIndexerServer(
-                      readService = readService,
-                      config = factory.indexerConfig(participantConfig, config),
-                      servicesExecutionContext = servicesExecutionContext,
-                      metrics = metrics,
-                      lfValueTranslationCache = lfValueTranslationCache,
-                    ).acquire().map(indexerHealth => healthChecks + ("indexer" -> indexerHealth))
+                    for {
+                      readLedger <- factory
+                        .readServiceOwner(config, participantConfig, sharedEngine)
+                        .acquire()
+                      readService = new TimedReadService(readLedger, metrics)
+                      healthChecksWithIndexer <- new StandaloneIndexerServer(
+                        readService = readService,
+                        config = factory.indexerConfig(participantConfig, config),
+                        servicesExecutionContext = servicesExecutionContext,
+                        metrics = metrics,
+                        lfValueTranslationCache = lfValueTranslationCache,
+                      ).acquire()
+                        .map(indexerHealth =>
+                          new HealthChecks(
+                            "read" -> readService,
+                            "indexer" -> indexerHealth,
+                          )
+                        )
+                    } yield {
+                      healthChecksWithIndexer
+                    }
                   case ParticipantRunMode.LedgerApiServer =>
-                    Resource.successful(healthChecks)
+                    Resource.successful(new HealthChecks())
```

Comment:
You sure we want no health checks here?

Reply:
Well, the health checks are used by the API server. In that case, we would add the write service health check there, but the read service would not even be initialized therefore this should be ok to have it empty.